### PR TITLE
DO NOT MERGE: do not show validation for props that have its default value

### PR DESCRIPTION
**Pull request by noodlemoodle**
https://nanopay.atlassian.net/browse/NP-3391

### CODE REVIEW

**Comment by jlhughes** [comment]
@noodlemoodle Can you add a screen shot please. Thanks.

**Comment by noodlemoodle** [comment]
sure so for example this postalcode field in brazil onboarding flow will not show validation when the value is the default empty string
Screen Shot 2021-02-02 at 2 52 24 AM
Screen Shot 2021-02-02 at 2 52 19 AM
Screen Shot 2021-02-02 at 2 52 10 AM

**Comment by jlhughes** [comment]
Tested with Treviso onboarding, and apart from inputting extra data, as there is no way to know what is really required, it works fine.

**Review by jlhughes — APPROVED** [review]

File: src/foam/u2/detail/SectionedDetailPropertyView.js [hunk 1]
```diff
@@ -244,7 +244,15 @@ foam.CLASS({
...
 
           var errorSlot = prop.validateObj && prop.validationTextVisible ?
-            this.data.slot(prop.validateObj) :
+            this.data.slot([[prop.name], function() {
+              var self = this;
+              var ax = self.cls_.getAxiomByName(prop.name);
+              if ( ax.isDefaultValue(self[prop.name]) ) return '';
+              if ( Array.isArray(ax.validateObj) ) 
+                return ax.validateObj[1].apply(self, ax.validateObj[0]);
+              else
```
Comment (kgrgreer) [inline src/foam/u2/detail/SectionedDetailPropertyView.js]: No need for 'else'.

File: src/foam/u2/detail/SectionedDetailPropertyView.js [hunk 2]
```diff
@@ -244,7 +244,15 @@ foam.CLASS({
         .add(this.slot(function(mode, prop, prop$label) {
 
           var errorSlot = prop.validateObj && prop.validationTextVisible ?
-            this.data.slot(prop.validateObj) :
+            this.data.slot([[prop.name], function() {
+              var self = this;
+              var ax = self.cls_.getAxiomByName(prop.name);
```
Comment (kgrgreer) [inline src/foam/u2/detail/SectionedDetailPropertyView.js]: What is the difference between 'ax' and 'prop'?

Comment (noodlemoodle) [reply src/foam/u2/detail/SectionedDetailPropertyView.js]: they are the same thing, i was doing something else wrong when prop didn't work.
updated, thanks

Comment (kgrgreer) [reply src/foam/u2/detail/SectionedDetailPropertyView.js]: Much better. Let's wait until after UX meeting before merging.

**Pull request closed without merging**